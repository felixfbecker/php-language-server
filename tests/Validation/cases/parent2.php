<?php

namespace MyNamespace;

class B {
    function b() {

    }
}

class A extends B {
    function a () {
        $a = PARENT::b();
    }
}
