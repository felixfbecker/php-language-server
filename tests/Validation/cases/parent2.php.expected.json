{
    "references": {
        "MyNamespace\\B": [
            "./_cases/parent2.php"
        ],
        "PARENT::b()": [
            "./_cases/parent2.php"
        ],
        "PARENT": [
            "./_cases/parent2.php"
        ]
    },
    "definitions": {
        "MyNamespace": {
            "extends": [],
            "isGlobal": true,
            "isStatic": false,
            "canBeInstantiated": false,
            "symbolInformation": {
                "name": "MyNamespace",
                "kind": 3,
                "location": {
                    "uri": "./_cases/parent2.php"
                },
                "containerName": ""
            },
            "type__class": "LanguageServer\\Tests\\ValidationTest",
            "type": null,
            "documentation": null
        },
        "MyNamespace\\B": {
            "extends": [],
            "isGlobal": true,
            "isStatic": false,
            "canBeInstantiated": true,
            "symbolInformation": {
                "name": "B",
                "kind": 5,
                "location": {
                    "uri": "./_cases/parent2.php"
                },
                "containerName": "MyNamespace"
            },
            "type__class": "LanguageServer\\Tests\\ValidationTest",
            "type": null,
            "documentation": null
        },
        "MyNamespace\\B->b()": {
            "extends": [],
            "isGlobal": false,
            "isStatic": false,
            "canBeInstantiated": false,
            "symbolInformation": {
                "name": "b",
                "kind": 6,
                "location": {
                    "uri": "./_cases/parent2.php"
                },
                "containerName": "MyNamespace\\B"
            },
            "type__class": "phpDocumentor\\Reflection\\Types\\Mixed",
            "type": {},
            "documentation": null
        },
        "MyNamespace\\A": {
            "extends": [
                "MyNamespace\\B"
            ],
            "isGlobal": true,
            "isStatic": false,
            "canBeInstantiated": true,
            "symbolInformation": {
                "name": "A",
                "kind": 5,
                "location": {
                    "uri": "./_cases/parent2.php"
                },
                "containerName": "MyNamespace"
            },
            "type__class": "LanguageServer\\Tests\\ValidationTest",
            "type": null,
            "documentation": null
        },
        "MyNamespace\\A->a()": {
            "extends": [],
            "isGlobal": false,
            "isStatic": false,
            "canBeInstantiated": false,
            "symbolInformation": {
                "name": "a",
                "kind": 6,
                "location": {
                    "uri": "./_cases/parent2.php"
                },
                "containerName": "MyNamespace\\A"
            },
            "type__class": "phpDocumentor\\Reflection\\Types\\Mixed",
            "type": {},
            "documentation": null
        }
    }
}